### PR TITLE
7 new contact UI

### PR DESCRIPTION
**Pull request by udai1931**

### CODE REVIEW

**Comment by yedhink** [comment]
@udai1931 Not sure why you are merging the PRs even after it's explicitly mentioned in the document to merge the PRs by yourself:

image

**Comment by udai1931** [comment]
@yedhink Actually I thought that after closing the PR I cannot assign you in that, but I was wrong : (
Merged all, Sorry for this.

**Comment by yedhink** [comment]
It's okay. Dont' worry about it.

File: app/javascript/src/components/Common/CreatePane.jsx [hunk 1]
```diff
@@ -2,24 +2,28 @@ import React from "react";
 
 import { Pane, Typography } from "neetoui";
 
-import Form from "./Form";
-
-export default function EditNotePane({
-  fetchNotes,
+export default function NewPane({
+  // fetchNotes,
```
Comment (yedhink) [inline app/javascript/src/components/Common/CreatePane.jsx:6]: Not sure why you've pushed a `comment` like this to a PR. What's the point of this comment?

File: app/javascript/src/components/Common/CreatePane.jsx [hunk 2]
```diff
@@ -2,24 +2,28 @@ import React from "react";
...
       <Pane.Header>
         <Typography style="h2" weight="semibold">
-          Edit Note
+          Create a New {PaneTitle}
         </Typography>
       </Pane.Header>
-      <Form isEdit onClose={onClose} refetch={fetchNotes} note={note} />
+      <FormComponent
+        onClose={onClose}
+        // refetch={fetchNotes}
```
Comment (yedhink) [inline app/javascript/src/components/Common/CreatePane.jsx:24]: Same as above

File: app/javascript/src/components/Common/DeleteAlert.jsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// @ts-nocheck
```
Comment (yedhink) [inline app/javascript/src/components/Common/DeleteAlert.jsx:1]: Remove this. In VSCode the errors will be there. Don't mind it.

File: app/javascript/src/components/Common/DeleteAlert.jsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import React from "react";
+
+import { Alert } from "neetoui";
+
+function DeleteAlert({ onClose, handleDelete }) {
+  return (
+    <div className="w-full">
+      <Alert
+        size="sm"
+        isOpen
```
Comment (yedhink) [inline app/javascript/src/components/Common/DeleteAlert.jsx:11]: Implicitly true boolean props ought to be mentioned as the starting props within the list of props.

https://www.bigbinary.com/react-best-practices/miscellaneous-tips#declaring-boolean-props

File: app/javascript/src/components/Common/Menubar.jsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+import { Search, Settings, Plus } from "neetoicons";
+import { Typography } from "neetoui";
+import { MenuBar } from "neetoui/layouts";
+
+function Menubar({ title, setCategory, options }) {
+  const [isSearchCollapsed, setIsSearchCollapsed] = useState(true);
+
+  return (
+    <MenuBar showMenu={true} title={title}>
```
Comment (yedhink) [inline app/javascript/src/components/Common/Menubar.jsx:12]: https://www.bigbinary.com/react-best-practices/miscellaneous-tips#declaring-boolean-props

File: app/javascript/src/components/Common/Menubar.jsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+      {options?.heads?.map((head, idx) => (
+        <MenuBar.Block
+          label={head.title}
+          key={idx}
```
Comment (yedhink) [inline app/javascript/src/components/Common/Menubar.jsx:16]: Index should be last resort:
https://medium.com/wesionary-team/using-index-as-a-key-is-an-anti-pattern-in-react-8e5db3aea3a9

File: app/javascript/src/components/Common/Menubar.jsx [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+          count={head.count}
+          onClick={() => setCategory(head.title)}
+        />
+      ))}
+
+      <MenuBar.SubTitle
+        iconProps={[
+          {
+            icon: Search,
+            onClick: () => setIsSearchCollapsed(!isSearchCollapsed),
```
Comment (yedhink) [inline app/javascript/src/components/Common/Menubar.jsx:26]: Please carefully read: https://www.bigbinary.com/react-best-practices/know-when-to-use-previous-state-for-state-update

File: app/javascript/src/components/Common/Sidebar/constants.js [hunk 8]
```diff
@@ -12,9 +13,17 @@ export const SIDENAV_LINKS = [
...
   {
     label: "Settings",
     to: "/settings",
-    icon: Settings,
+    icon: Download,
   },
 ];
+
+export const AVATAR_IMAGE_URL =
+  " https://i.pinimg.com/originals/51/f6/fb/51f6fb256629fc755b8870c801092942.png";
```
Comment (yedhink) [inline app/javascript/src/components/Common/Sidebar/constants.js:29]: Why is this split into two lines? Was this auto-formatted? If not, make it a single line declaration.

File: app/javascript/src/components/Common/Sidebar/index.jsx [hunk 9]
```diff
@@ -1,67 +1,49 @@
...
 
   const bottomLinks = [
     {
       label: "My Profile",
-      onClick: () => history.push(PROFILE_PATH, { resetTab: true }),
-    },
-    {
-      label: "Change Password",
-      onClick: () => history.push(CHANGE_PASSWORD_PATH, { resetTab: true }),
+      // onClick: () => history.push(PROFILE_PATH, { resetTab: true }),
```
Comment (yedhink) [inline app/javascript/src/components/Common/Sidebar/index.jsx:14]: Again. Go through the whole PR and remove these useless comments.

File: app/javascript/src/components/Contacts/contacts.js [hunk 10]
```diff
@@ -0,0 +1,149 @@
+export const contacts = [
+  {
+    id: 1,
+    name: "Udai Gupta",
+    role: "Engineer",
+    email: "udai.gupta@bigbinary.com",
```
Comment (yedhink) [inline app/javascript/src/components/Contacts/contacts.js:6]: Never use real world names or emails in seed data. Use random emails which ends in `example.com`.

Also in BigBinary we stick onto certain personas: https://handbook.neetokb.com/articles/personas

File: app/javascript/src/components/Notes/index.jsx [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+
+import DeleteAlert from "../Common/DeleteAlert";
+
+const Notes = () => {
+  const [showNewNotePane, setShowNewNotePane] = useState(false);
+  const [searchTerm, setSearchTerm] = useState("");
+  const [notes, setNotes] = useState([]);
+  const [category, setCategory] = useState("All");
+  const [showDeleteAlert, setShowDeleteAlert] = useState(false);
+  const [, setSelectedNoteIds] = useState(null);
```
Comment (yedhink) [inline app/javascript/src/components/Notes/index.jsx:25]: What's the point of declaring a hook setter without a state? Couldn't you use a variable here rather? What are the caveats?

**Review by yedhink — COMMENTED** [review]
@udai1931 Added few comments. You can fix them in a separate PR or a PR that you are currently working on, in this repo.

PS: I haven't pinpointed all spots within this PR where similar changes ought to be made. It's your responsibility to go through the PR and figure out where all similar changes ought to be made based on the comments I have added.

Lemme know if you have any queries.

File: app/javascript/src/components/Common/CreatePane.jsx (hunk 1 above)
Comment (udai1931) [reply app/javascript/src/components/Common/CreatePane.jsx:6]: It was just to make an idea of how this component will work while building it. Will remove all of them.

File: app/javascript/src/components/Notes/index.jsx (hunk 11 above)
Comment (udai1931) [reply app/javascript/src/components/Notes/index.jsx:25]: When I declared that variable, it was it was giving the error of variable not used is pre commit check as no actual request was made using that Id

Comment (yedhink) [reply app/javascript/src/components/Notes/index.jsx:25]: If selected note ids aren't used then what's the point of this state at all?

Comment (udai1931) [reply app/javascript/src/components/Notes/index.jsx:25]: Yes, as no request is made its of no use. Removed it. Actually added in the start just to understand the functionality.

File: app/javascript/src/components/Common/DeleteAlert.jsx (hunk 3 above)
Comment (udai1931) [reply app/javascript/src/components/Common/DeleteAlert.jsx:1]: Any proper solution to remove those errors as there are many of them ?

**Pull request merged**